packages/microcosm/src/data.js

```diff
@@ -117,49 +129,48 @@ export function get(object: ?Object, keyPath: *, fallback?: *) {
 
 /**
  * Merge any number of objects into a provided object.
+ * $FlowFixMe This function enumerates on variable arguments
  */
-export function merge(...args: Array<?Object>): Object {
-  let copy = null
-  let subject = null
-
-  for (var i = 0, len = args.length; i < len; i++) {
-    if (isObject(args[i]) === false) {
-      continue
-    }
+export function merge(): Object {
+  let cloned = false
+  let copy = EMPTY_OBJECT
 
-    copy = copy || args[i] || {}
+  for (var i = 0, len = arguments.length; i < len; i++) {
+    var next = isObject(arguments[i]) ? arguments[i] : null
 
-    subject = subject || copy
+    if (copy === EMPTY_OBJECT) {
+      copy = next || EMPTY_OBJECT
+    }
 
-    var next = args[i]
     for (var key in next) {
       if (copy[key] !== next[key]) {
-        if (copy === subject) {
-          copy = clone(subject)
+        if (cloned === false) {
+          copy = clone(copy)
+          cloned = true
         }
 
         copy[key] = next[key]
       }
     }
   }
 
-  return copy || {}
+  return copy == EMPTY_OBJECT ? {} : copy
```

Comment:
@efatsi I ended up needing this check, but we now have coverage for it. 

Logically speaking, `EMPTY_OBJECT` is the same as `null` in this case. It's just that it isn't actually null. It's a null object I guess. I like this better, and it makes Flow happier.

What do you think?

Reply:
no strong opinions here from me

Reply:
no strong opinions here from me

Reply:
Cool. I'll ship this. Thanks!